R/plotconfiguration-timeprofile.R

```diff
@@ -1,42 +1,110 @@
 #' @title TimeProfilePlotConfiguration
 #' @docType class
-#' @description  Plot Configuration for PKRatio
+#' @description  Plot Configuration for Time Profile Plots
 #' @export
 TimeProfilePlotConfiguration <- R6::R6Class(
   "TimeProfilePlotConfiguration",
   inherit = PlotConfiguration,
+
   public = list(
-    LLOQLineProperties = NULL,
+    LLOQLinesProperties = NULL,
+    colorGrouping = NULL,
+    linetypeGrouping = NULL,
 
-    initialize = function(title = "Time Profile Plot",
-                              LLOQLineProperties = NULL,
-                              dataMapping = NULL,
+    initialize = function(LLOQLinesProperties = tlfEnv$currentTheme$LLOQLinesProperties,
+                              colorGrouping = NULL,
+                              linetypeGrouping = NULL,
+                              title = "Time Profile Plot",
+                              subtitle = paste("Date:", format(Sys.Date(), "%y-%m-%d")),
+                              xlabel = NULL,
+                              ylabel = NULL,
+                              watermark = tlfEnv$currentTheme$watermarkText,
+                              data = NULL,
                               metaData = NULL,
+                              dataMapping = NULL,
                               ...) {
-      timeProfileDataMapping <- NULL
-      timeProfileMetaData <- NULL
-      if (!is.null(dataMapping)) {
-        timeProfileDataMapping <- dataMapping$simulationMapping
-        timeProfileMetaData <- metaData[[dataMapping$simulationSets]]
-      }
-
       super$initialize(
         title = title,
-        metaData = timeProfileMetaData,
-        dataMapping = timeProfileDataMapping,
-        ...
+        subtitle = subtitle,
+        xlabel = xlabel,
+        ylabel = ylabel,
+        watermark = watermark,
+        data = data,
+        metaData = metaData,
+        dataMapping = dataMapping
       )
+
+      self$LLOQLinesProperties <- LLOQLinesProperties
+
+      if (!is.null(dataMapping$groupings)) {
+        self$colorGrouping <- dataMapping$groupings[["color"]]$groupingName[[1]]

```

Comment:
It feels that the groupings should be an object with properties color, linetype etc.. Let's discuss

Reply:
I have reshaped the grouping quite a few times.
Abdullah's presentation showed there is far more potential for this class compared to what I was expecting at the beginning. It indeed deserve a discussion to figure what we want from this class and consequently how we want it to behave for the API.